### ISSUE TITLE
Items in chat are incorrectly parsed

### ISSUE THREAD

**Issue by AwesomeFart** [domialex/Sidekick#82]
Most of the uniques wont show price when CTRL+D from chat.

There is just a blank window for a split second. Some of them work tho. Had no Issues to CTRL+D from Stash or Iventory.

**Comment by AwesomeFart** [comment]
![bug](https://user-images.githubusercontent.com/59807858/72226392-eb738280-3590-11ea-8d80-0785af6ed3a3.png)

**Comment by domialex** [comment]
Every item from the chat appears to have some annotations in their name used for localisation :
FR - `<<set:F>><<set:S>><<set:FS>>Cœur de Kaom`
EN - `<<set:MS>><<set:M>><<set:S>>Kaom's Heart`
Flask - FR `<<set:M>><<set:S>><<set:MS>>Flacon divin de vie <if:MS>{Catalysé}<elif:FS>{Catalysée}<elif:MP>{Catalysés}<elif:FP>{Catalysées} de la Cicatrisation`

Gotta use a regex here :*(

**Comment by domialex** [comment]
From my current understanding, the first  `<<set:X>>` sets the gender (M for masculine, F for feminine) and number (S for singular, P for plural)
So: `<<set:MP>><<set:M>><<set:P>>Clous<if:MS>{ bestial}<elif:FS>{ bestiale}<elif:NS>{NONEXISTENT}<elif:MP>{ bestiaux}<elif:FP>{ bestiales}<elif:NP>{NONEXISTENT}`
Would give:
`Clous bestiaux`

**Comment by cricket1650** [comment]
I’ll take this on. It should just be a quick regex tokenizer and a parser.

**Comment by domialex** [comment]
Ok then I can give you the horror I was messing around with.
```c#
public static string CleanName(string input)
{
    if (!input.Contains("<<"))
    {
        return input;
    }

    // Get setters.
    var setterRegex = new Regex(@"<<set:(?<KEY>.*?)>>");
    var settersMatch = setterRegex.Matches(input);
    var setters = settersMatch.Select(x => x.Groups["KEY"].Value).ToList();

    // Get conditions.
    var conditionRegex = new Regex(@"<(?:if|elif):(?<KEY>.*?)>{(?<VALUE>.*?)}");
    var conditionsMatch = conditionRegex.Matches(input);

    var match = conditionsMatch.SingleOrDefault(x => setters.Contains(x.Groups["KEY"].Value))?.Groups["VALUE"]?.Value;
    if (!string.IsNullOrWhiteSpace(match))
    {
        input = conditionRegex.Replace(input, (x) => setters.Contains(x.Groups["KEY"].Value) ? x.Groups["VALUE"].Value : string.Empty);
    }

    var lastInstance = input.LastIndexOf(">>");
    return input.Substring(lastInstance + 2);
}
```
I'm still not sure if a name can have multiple <<if:XX>> or just one.

**Comment by domialex** [comment]
Implemented with #112.
